src/merge_iterator.rs

```diff
@@ -41,6 +42,53 @@ impl<T1: KeyValueIterator, T2: KeyValueIterator> TwoMergeIterator<T1, T2> {
     }
 }
 
+impl<T1, T2> TwoMergeIterator<T1, T2>
+where
+    T1: KeyValueIterator + SeekToKey,
+    T2: KeyValueIterator + SeekToKey,
+{
+    async fn seek1(&mut self, next_key: &Bytes) -> Result<(), SlateDBError> {
+        match &self.iterator1.1 {
+            None => Ok(()),
```

Comment:
I guess it should probably be an error. It might require some additional bookkeeping on the iterator implementations though to keep track of the last returned key. Let me see what this looks like..